### PR TITLE
[RFC] Add ISSUE_TEMPLATE.md

### PR DESCRIPTION
**Pull request by mhinz**
There is a new Github feature. Please read the following links first.
- https://help.github.com/articles/creating-an-issue-template-for-your-repository/
- https://github.com/blog/2111-issue-and-pull-request-templates

The actual PR is just a possible suggestion and there are probably lots of changes needed. (E.g. "How was Neovim installed? (Homebrew, distro package manager etc.))

Please suggest your favourite template in the comments!

### CODE REVIEW

File: ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+**Neovim version**:
+**Operating system**:
+**Terminal emulator**:
```
Comment (justinmk) [inline ISSUE_TEMPLATE.md]: rendered markdown shows these 3 lines on 1 line

**Comment by mhinz** [comment]
Ah, I like the idea of a minimal vimrc since people often don't know what comes from Neovim and what comes from a plugin. We could add something like the following to the "Steps to reproduce" section:

- [ ] Reproducable with the [minimal config](https://github.com/neovim/neovim/blob/master/mini-init.vim)

**Comment by Shougo** [comment]
@mhinz It should be in `.github/ISSUE_TEMPLATE.md`.

**Comment by mhinz** [comment]
Okay, I moved `ISSUE_TEMPLATE.md` and `CONTRIBUTING.md` to `.github`.

**Comment by Shougo** [comment]
:+1:

**Comment by justinmk** [comment]
Sorry, I actually think we don't need `.github/` folder. Both files are not really github-specific (and I don't think we should add a "PR template").

**Comment by Shougo** [comment]
@justinmk the issue template is supported in others?

**Comment by Shougo** [comment]
I think CONTRIBUTING.md should be in the top directory and
ISSUE_TEMPLATE.md should be in .github directory...

**Comment by mhinz** [comment]
I agree that `CONTRIBUTING` is quite a common file in general and should stay in root. I'm unpassionate about whether `ISSUE_TEMPLATE.md` or `.github/ISSUE_TEMPLATE.md`, but I understand the sentiment that the latter might not be needed if it only contains a single file anyway.

**Comment by mhinz** [comment]
I say we just merge it and see if people embrace it?

**Pull request merged**